### ISSUE TITLE
[iOS][KeyboardAvoidingView] Using KeyboardAvoidingView inside a ScrollView causes the app the crash when focusing an input

### ISSUE THREAD

**Issue by slauzinho** [facebook/react-native#42939]
### Description

When `KeyboardAvoidingView` inside a `ScrollView` and focus a `TextInput` the app crashes (goes extremely slow).

This seems to only happen if the `KeyboardAvoidingView` needs to trigger the animation, if the ScrollView is small it works fine.

Testing with react native `0.72.9` the issue doesn't seem to happen

### Steps to reproduce

1. install the app `yarn ios`
2. Using the simulator make sure the keyboard shows when entering a text input
3. Focus a text input and start typing on the keyboard
4. Dismiss the keyboard / click outside of the input
5. You will notice the app has now hung 

### React Native Version

0.73.4

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.2.1
  CPU: (10) arm64 Apple M1 Max
  Memory: 127.84 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.18.2
    path: ~/.nvm/versions/node/v18.18.2/bin/node
  Yarn:
    version: 1.22.21
    path: ~/.nvm/versions/node/v18.18.2/bin/yarn
  npm:
    version: 9.8.1
    path: ~/.nvm/versions/node/v18.18.2/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.14.3
    path: /usr/local/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.2
      - iOS 17.2
      - macOS 14.2
      - tvOS 17.2
      - visionOS 1.0
      - watchOS 10.2
  Android SDK: Not Found
IDEs:
  Android Studio: 2021.1 AI-211.7628.21.2111.8309675
  Xcode:
    version: 15.2/15C500b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.9
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.4
    wanted: 0.73.4
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```
```


### Stacktrace or Logs

```text
Excessive number of pending callbacks: 501. Some pending callbacks that might have leaked by never being called from native code: {"235":{"module":"UIManager","method":"configureNextLayoutAnimation"},"238":{"module":"UIManager","method":"configureNextLayoutAnimation"},"241":{"module":"UIManager","method":"configureNextLayoutAnimation"},"244":{"module":"UIManager","method":"configureNextLayoutAnimation"},"247":{"module":"UIManager","method":"configureNextLayoutAnimation"},"250":{"module":"UIManager","method":"configureNextLayoutAnimation"},"253":{"module":"UIManager","method":"configureNextLayoutAnimation"},"256":{"module":"UIManager","method":"configureNextLayoutAnimation"},"259":{"module":"UIManager","method":"configureNextLayoutAnimation"},"262":{"module":"UIManager","method":"configureNextLayoutAnimation"},"265":{"module":"UIManager","method":"configureNextLayoutAnimation"},"268":{"module":"UIManager","method":"configureNextLayoutAnimation"},"271":{"module":"UIManager","method":"configureNextLayoutAnimation"},"274":{"module":"UIManager","method":"configureNextLayoutAnimation"},"277":{"module":"UIManager","method":"configureNextLayoutAnimation"},"280":{"module":"UIManager","method":"configureNextLayoutAnimation"},"283":{"module":"UIManager","method":"configureNextLayoutAnimation"},"286":{"module":"UIManager","method":"configureNextLayoutAnimation"},"289":{"module":"UIManager","method":"configureNextLayoutAnimation"},"292":{"module":"UIManager","method":"configureNextLayoutAnimation"},"295":{"module":"UIManager","method":"configureNextLayoutAnimation"},"298":{"module":"UIManager","method":"configureNextLayoutAnimation"},"301":{"module":"UIManager","method":"configureNextLayoutAnimation"},"304":{"module":"UIManager","method":"configureNextLayoutAnimation"},"307":{"module":"UIManager","method":"configureNextLayoutAnimation"},"310":{"module":"UIManager","method":"configureNextLayoutAnimation"},"313":{"module":"UIManager","method":"configureNextLayoutAnimation"},"316":{"module":"UIManager","method":"configureNextLayoutAnimation"},"319":{"module":"UIManager","method":"configureNextLayoutAnimation"},"322":{"module":"UIManager","method":"configureNextLayoutAnimation"},"325":{"module":"UIManager","method":"configureNextLayoutAnimation"},"328":{"module":"UIManager","method":"configureNextLayoutAnimation"},"331":{"module":"UIManager","method":"configureNextLayoutAnimation"},"334":{"module":"UIManager","method":"configureNextLayoutAnimation"},"337":{"module":"UIManager","method":"configureNextLayoutAnimation"},"340":{"module":"UIManager","method":"configureNextLayoutAnimation"},"343":{"module":"UIManager","method":"configureNextLayoutAnimation"},"346":{"module":"UIManager","method":"configureNextLayoutAnimation"},"349":{"module":"UIManager","method":"configureNextLayoutAnimation"},"352":{"module":"UIManager","method":"configureNextLayoutAnimation"},"355":{"module":"UIManager","method":"configureNextLayoutAnimation"},"358":{"module":"UIManager","method":"configureNextLayoutAnimation"},"361":{"module":"UIManager","method":"configureNextLayoutAnimation"},"364":{"module":"UIManager","method":"configureNextLayoutAnimation"},"367":{"module":"UIManager","method":"configureNextLayoutAnimation"},"370":{"module":"UIManager","method":"configureNextLayoutAnimation"},"373":{"module":"UIManager","method":"configureNextLayoutAnimation"},"376":{"module":"UIManager","method":"configureNextLayoutAnimation"},"379":{"module":"UIManager","method":"configureNextLayoutAnimation"},"382":{"module":"UIManager","method":"configureNextLayoutAnimation"},"...(truncated keys)...":451}
```
```


### Reproducer

https://github.com/slauzinho/KeyboardAvoidingViewExample

### Screenshots and Videos



https://github.com/facebook/react-native/assets/445345/72aa333e-f55a-446f-aea5-6b7615b8a4ca

**Comment by kokosky93** [comment]
Upgraded to react native 0.73.4. All text inputs are super slow. Application crashes on IOS when focusing on an input.

**Comment by Hasanraza786** [comment]
Kindly check this key in scrollview 
overScrollMode='never'
@kokosky93

**Comment by slauzinho** [comment]
> Kindly check this key in scrollview overScrollMode='never' @kokosky93

`overScrollMode` is an Android prop shouldn't really matter in this case

**Comment by kokosky93** [comment]
That's true. Setting overScrollMode didn't help.

**Comment by zaferatli** [comment]
i dont think its crash, its loop cause mis-use of element. in this case keyboardawarescroll in scrollview; when keyboard is open its trigger to keyboardawarescroll to extend bottom height and put keyboard down and thats circle trigger to scrollview because layout change and also scrollview extends layout and keyboardawarescroll again try to put keyboard to bottom so extend layout and again triggered scrollview.

Even i prevent this loop keyboardawarescroll its not working properly, can't arrange keyboard position and content because scrollview.

Related part KeyboardAvoidingView.js:

```
   // update bottom height for the first time or when the height is changed
    if (!oldFrame || oldFrame.height !== this._frame.height) {
      await this._updateBottomIfNecessary();
    }
```

**Comment by tomerh2001** [comment]
Any updates? Also experiencing this - v0.73.4

**Comment by mpringlehlk** [comment]
Like @kokosky93 , my workaround  was to swap the component order to make the `KeyboardAvoidingView` the parent element of the `ScrollView`, then add a new child `View` for applying container styles that were previously on the `KeyboardAvoidingView`.

So far, everything seems to be working as intended.

I also observed that this issue only affected distributions running on real iOS devices. Not reproducible in either ExpoGo or Android builds.

**Comment by marvinfok** [comment]
> Experiencing the same issue here! RN v0.73.4 as well. The issue indeed seems to be related to being inside a `ScrollView`, and on top having the behaviour set to `padding`, like so: `<KeyboardAvoidingView behavior="padding">`
> 
> When setting the behaviour to `position`, there is no problem. ` <KeyboardAvoidingView behavior="position">`
> 
> Edit: added more details

Thanks buddy! Works! Yes, must change behavior to ="position", else will not scroll.

**Comment by jasenpashov** [comment]
Thanks

**Comment by janoslc** [comment]
Experienced the same thing, I had `<KeyboardAvoidingView/>` inside of a `<SafeAreaView/>`. It works on the iOS Emulator, on Android simulator and device, it only "freezes" on real iOS device.  I'm following this conversation now.

**Comment by ericTrayt** [comment]
Experiencing the same thing here as well. 0.73.6

**Comment by ajakka** [comment]
Any updates in regards to this issue?

**Comment by shahidrogers** [comment]
I'm facing this same issue too, using `<KeyboardAvoidingView/>` inside of a `<SafeAreaView/>`.

It only freezes on a real iOS device - can't reproduce on simulator/android.

"react-native": "0.73.7"

**Comment by slauzinho** [comment]
Still happens with react native `0.74.1`

**Comment by joshzeldin** [comment]
I think I’m experiencing something similar with TextInput inside of a KeyboardAvoidingView on react native 0.74.1.

**Comment by olofd** [comment]
Same here.. 0.74.1

**Comment by safee-cases** [comment]
Same issue here. Problem seems to appear only on "small" device: working perfectly on iPhone 11, crashing on iPhone 13 mini. 

I'm currently trying to implement this PR in local before it get merged:

https://github.com/facebook/react-native/pull/44652

**Comment by yummyelin** [comment]
> Same issue here. Problem seems to appear only on "small" device: working perfectly on iPhone 11, crashing on iPhone 13 mini.
> 
> I'm currently trying to implement this PR in local before it get merged:
> 
> #44652

tried this change locally, seems not working for me

**Comment by patrickkeenan** [comment]
Glad this is being fixed. Its super frustrating because it only appears when uploaded to test flight. I couldn't reproduce the error in my development build.

**Comment by JB712** [comment]
I think I'm facing the same issue

> Same issue here. Problem seems to appear only on "small" device: working perfectly on iPhone 11, crashing on iPhone 13 mini.
> 
> I'm currently trying to implement this PR in local before it get merged:
> 
> #44652

I think I'm facing the same issue with an iPad (10th gen) and an iPhone 11 so not only small devices. Besides, I have an iPhone 15 that doesn't encounter this bug.
I'm working with react-native@0.73.9

**Comment by Jmzp** [comment]
I have the same problem in iOS with RN .73.9.

**Comment by maratkarelov** [comment]
> I have the same problem in iOS with RN .73.9.

remove KeyboardAvoidingView

**Comment by DnEgorWeb** [comment]
Hey guys, anyone knows alternative to KeyboardAvoidingView?

**Comment by fz6m** [comment]
move `ScrollView` inside `KeyboardAvoidingView` , it works for me

**Comment by fstefanello** [comment]
I found a simple example where I can reproduce a similar issue on the Android emulator. I'm not sure, but I believe the problem is related.

The issue occurs when the position of the KeyboardAvoidingView is above the size of the keyboard. This causes the KeyboardAvoidingView component to keep increasing the padding indefinitely.

  ```
render() {
    return (
      <View>
        <View style={{borderWidth: 1, height: DEVICE_HEIGHT - 200}}/>
        <KeyboardAvoidingView behavior='padding'>
          <TextInput style={{borderWidth: 1}} />
        </KeyboardAvoidingView>
      </View>
    );
  }
```

**Comment by matinzd** [comment]
This is also happening for our users on some real iOS devices. I am both using safe area and KeyboardAvoidingView from react native.

Going to replace it with [keyboard controller](https://github.com/kirillzyusko/react-native-keyboard-controller) and remove safe area view to use hook instead. Going to release the app to see if that fixes the issue.

It's very hard to debug this issue since I couldn't reproduce it myself.

@cipolleschi Maybe you can help us here? Feel like a regression that has been around for while in RN.

**Comment by matinzd** [comment]
Removing `SafeAreaView` and using keyboard controller fixed the issue and we verified it with one of our customers. I still can't reproduce it on my end.

**Comment by kubiisimsek** [comment]
Hi guys

I have the same problem with rn 0.73.9 on iOS real device in production and scheme build config:Release on simulator. It worried me but solved.

 ```js
 <ScrollView>
     <KeyboardAvoidingView behavior={Platform.OS == "ios" ? "paddding":null}>
     ...
     </KeyboardAvoidingView>
 </ScrollView>
 ```

 If KeyboardAvoidingView inside the ScrollView with prop behavior="padding" in iOS,screen is freezes.
 I change behavior padding to position for ios it worked but screen bounced when typing.
 
 Using it belows everything worked for me.
  ```js
  <KeyboardAvoidingView behavior={Platform.OS == "ios" ? "paddding":null}>
    <ScrollView>
     ...
     </ScrollView>
 </KeyboardAvoidingView>
 ```
 
Some said above that it was caused by SafeAreaView. My SafeAreaView defined in App.js. In my project, it's not because of it.
 
 I hope it'll fixed.

**Comment by cipolleschi** [comment]
Hello there, I'm sorry that you are encountering this issue.
I see that this is happening with the Old Architecture. Is it happening also with the New Architecture?

At the moment we are hyper focused on rolling out the New Architecture, so issues for the Old Architecture only has lower priority as they will move capacity away from our final goal.

**Comment by omgaz** [comment]
@cipolleschi We're seeing this issue too across old and new devices; we've just upgraded to `"react-native": "0.73.8"` and are only seeing it on 'real' ios devices.

![Screenshot 2024-09-03 at 14 34 13](https://github.com/user-attachments/assets/1b86d715-7425-420b-a99d-354e71726cad)

Similar to https://github.com/facebook/react-native/issues/42939#issuecomment-2071272461, we're getting it on `SafeAreaView`; we'll see if we can flip it too and see what happens.

**Comment by cipolleschi** [comment]
I'm looking into this, and I see that the app get blocked and eventually crashes.
However, this bug is only happening in the Old Architecture, so sadly I can't spend more time on this as we should focus on the New Architecture.

To unblock on this, I actually think that you should not use the `<KeyboardAvoidingView>` component inside a scrollview. If what you want to scroll is the scrollview, the `<ScrollView>` component has the prop [`automaticallyAdjustKeyboardInsets`](https://reactnative.dev/docs/scrollview#automaticallyadjustkeyboardinsets-ios) that should fix the issue on the Old Architecture.
It should also work in the New Architecture.

**Comment by mkhotib20** [comment]
> move `ScrollView` inside `KeyboardAvoidingView` , it works for me

me too, you can move it. it works

**Comment by matinzd** [comment]
@cipolleschi How did you reproduce it?

**Comment by cipolleschi** [comment]
@matinzd with the reproducer attached to the issue:
https://github.com/slauzinho/KeyboardAvoidingViewExample

**Comment by johnnywang** [comment]
Been debugging an issue with our app freezing on certain screens only in real iOS 17.5+ devices, and finally narrowed it down to this same issue here that people have been reporting.

Some details:
- We have a KeyboardAvoidingView inside a SafeAreaView
- Only happens when the behavior is set to "padding" (though anything else breaks the UI)
- Only seems to happen (100% consistently) on real iOS 17 (possibly only 17.5+) devices
- Does not happen / cannot repro on emulators
- Ejected Expo 51.0.38
- React Native 0.75.4

**Comment by cipolleschi** [comment]
@johnnywang Thanks for your message, I empathize with your situation.

Can you provide a reproducer for your scenario? This specific issue is related to KeyboardAvoidingView inside a scrollview, and that's something that should not be needed.

> @cipolleschi while I understand the prioritization of the new architecture right now, I'd argue that there are probably a good number of teams and products still using older versions of RN. Even for those of us who are on the newer versions where we can opt into the new architecture, there are a lot of third party packages and dependencies that don't support the new architecture yet, which leaves us in a tough spot because "just upgrade to the new architecture" is not a solution. (And as a few others have also reported, it's not just a ScrollView / SafeArea issue)

I know that there are team stuck on older version of React Native, we are doing our best to try and fix what we can, but we also need to focus our limited energies in pushing the ecosystem forward. 
With 0.76, 926 libraries of 1538 libraries tracked by the [reactnative.directory](https://reactnative.directory/) website are on the new architecture: 60.2%.
Thanks to the interop layers, we expect that also all the other libraries should work  fine with the New Architecture with no changes, therefore the suggestion of moving to the New Architecture remains.

**Comment by matinzd** [comment]
> Even for those of us who are on the newer versions where we can opt into the new architecture, there are a lot of third party packages and dependencies that don't support the new architecture yet, which leaves us in a tough spot because "just upgrade to the new architecture" is not a solution. (And as a few others have also reported, it's not just a ScrollView / SafeArea issue)

I can also second this. I also removed the use of SafeAreaView but I am still getting complaints from the users. I have tried migrating to new arch multiple times but there has been some bugs with some libraries and I still have to opt out due to these reasons. I always keep all the libs up-to-date but still no luck with the new architecture. I now tried to move away from controlled input and will be using uncontrolled input in our app to see if that fixes the issue. 
I will give you more update on that when we release the app and get some adoption.

**Comment by johnnywang** [comment]
> Can you provide a reproducer for your scenario? This specific issue is related to KeyboardAvoidingView inside a scrollview, and that's something that should not be needed.

I believe a few other commenters have mentioned that they're also experiencing this when used inside a SafeAreaView, although I was noticing this happening even without that. I'm not sure if I'll be able to come up with a repro, but I'll link here if I do. My guess is that it's some combination of:

- Navigating to a new screen that has an input field and a KeyboardAvoidingView on it
- Only on real iOS devices in the iOS 17 range (probably 17.5+)

I've been able to get around the problem for now by following what others have done and moving over to keyboard-controller's KAV + using the safe area hook instead, but it's admittedly a little frustrating since this is an unexpected/unnecessary migration

**Comment by cipolleschi** [comment]
I agree with the frustration and we are committed to make it easier. However, we need some reproducer to be able to understand the problem and to find the root cause for it. You can see how saying that "KAV doesn't work when in a Safe Area View" does not really helps... Especially because there are mentions with the problem arise on modal or upon Navigation. react Native does not provide primitives to navigate among screens, so we can't exclude that the problem might be in the navigation library you are using.

That's why the reproducer is paramount to isolate the problem and verify that is a core issue rather than a 3p issue that ripples to core.

**Comment by mobinni** [comment]
Hi there, I think this issue deserves a lot more attention and a remediation or patch.

We've been experiencing this in various places in our app since upgrading. We have a relatively large codebase with various patterns of keyboard avoiding view.

Although swapping to `behavior="padding"` ensures that it won't hang and break what has actually happened is we've upgraded to 0.73 is we 

Regression testing allows us to catch some of the places, this bug isn't on all iOS devices, it's intermittent at best and not everyone is affected so we can't be confident.

Basically a core pattern that has worked forever in react native is now broken, we are not ready to move to the new architecture and was not present on 0.72.+

I'm not sure what has changed between 72 and 73 in the keyboard avoiding view logic for iOS and I am more than happy to help track down and remediate the root cause with some guidance

**Comment by cipolleschi** [comment]
Thanks @mobinni for the stable reproducer.
We will definitely have a look at it. To set expectation, I'll be off for the next week and @cortinico will be back on Thursday, but it will have an higher priority when we are back.

**Comment by mobinni** [comment]
@cipolleschi Thanks for getting back to me 🙏 to echo @omgaz enjoy your time off and thank you very much for being so active! We have a remediation in place, I would love to help more here if I can

**Comment by mobinni** [comment]
Adding this to KeyboardAvoidingView stops the release bug:

```
    if (enabled && duration && easing && Platform.OS !== 'ios') {
      LayoutAnimation.configureNext({
        // We have to pass the duration equal to minimal accepted duration defined here: RCTLayoutAnimation.m
        duration: duration > 10 ? duration : 10,
        update: {
          duration: duration > 10 ? duration : 10,
          type: LayoutAnimation.Types[easing] || 'keyboard',
        },
      });
    }
```
In testing this I did not see any immediate regressions in the behaviour of KBAV

**Comment by matinzd** [comment]
> Adding this to KeyboardAvoidingView stops the release bug:
> 
> 
> 
> ```
> 
>     if (enabled && duration && easing && Platform.OS !== 'ios') { // <== this line is different
> 
>       LayoutAnimation.configureNext({
> 
>         // We have to pass the duration equal to minimal accepted duration defined here: RCTLayoutAnimation.m
> 
>         duration: duration > 10 ? duration : 10,
> 
>         update: {
> 
>           duration: duration > 10 ? duration : 10,
> 
>           type: LayoutAnimation.Types[easing] || 'keyboard',
> 
>         },
> 
>       });
> 
>     }
> 
> ```
> 
> In testing this I did not see any immediate regressions in the behaviour of KBAV

Sorry but how does this code would help when the condition specifically says `OS != iOS`? This issue specifically says this is happening on iOS.

**Comment by mobinni** [comment]
@matinzd sorry I should have been more clear I added the line `Platform.OS !== 'ios'` to stop the call to `LayoutAnimation.configureNext` which causes the app to hang

**Comment by mobinni** [comment]
@cipolleschi As said in the comment it's only on 0.73.+, I guess if we're not looking to patch it for that version then there is nothing to do

**Comment by cipolleschi** [comment]
Yes, 0.73 is out of support now. I'm glad that this is fixed in versions > 0.73.

**Comment by dan-trewin** [comment]
@cipolleschi  In my experience and seemingly others based on their comments in this github issue, this crash/freeze still happens when new architecture is off on both react-native v0.75 and v0.76

**Comment by cipolleschi** [comment]
A great reason to migrate to the New Architecture! 🤣 

Jokes aside, thanks for the clarification, I'll try and investigate it further. I'm reopening the issue.

**Comment by anta-semenov** [comment]
Can confirm that we have the same issue with old architecture on 0.76.6.

As a workaround before switching to new architecture it's possible to use this old library https://www.npmjs.com/package/react-native-keyboard-aware-scroll-view that do the same. It doesn't have such issue

**Comment by mobinni** [comment]
@anta-semenov another temporary remediation is to always use `padding` on iOS

**Comment by flemton** [comment]
Try using the new architecture, my crashes vanished after switching

**Comment by ftroisi** [comment]
I was having this issue with Expo SDK 52 and both ReactNative 0.76.9 and 0.77.2. For me the solution was to switch to the new architecture (setting `"newArchEnabled": true` in `app.json`)

**Comment by ics-ankit** [comment]
i have same issue on Expo SDK ~52.0.46 and both RN 0.73 and 0.76.9 . also i have not add the newArchEnable: true in app.json

i am share my code 
app frozen and when reopen the app it will crash in ios `iPhone 15 Pro Max +1iOS 18.5`
`    <SafeAreaView style={[styles.safeArea]}>
      <TouchableWithoutFeedback onPress={() => { Keyboard.dismiss() }}>
        <View
          style={{
            flex: 1,
            paddingHorizontal: 20,
          }}>
          <View style={[styles.flex]}>
            <View style={{ marginTop: 30 }}>
              <FontAwesome6
                name={'users'}
                size={48}
                color={'#E898FE'}
                alignSelf={'center'}
              />
            </View>

            <Text style={{ ...styles.groupSuggetionText, marginTop: 20 }}>
              If you do not see your city, please search the full name of the
              state or province you live in and select the region closest to you
              on the previous screen. All groups have a long range to protect as
              many women as possible.
            </Text>

            <Text style={{ ...styles.groupSuggetionText }}>
              If you’d like to suggest a region that you want a dedicated group
              for, please let us know below.
            </Text>

            <View
              style={[
                styles.searchView,
                { backgroundColor: 'white', marginBottom: 5 },
              ]}>
              <FontAwesome
                name="search"
                size={18}
                color={colorSet.secondaryText}
                style={{ marginHorizontal: 10 }}
              />
              <TextInput
                allowFontScaling={false}
                placeholder="Enter city name...."
                placeholderTextColor={colorSet.secondaryText}
                value={text}
                onChangeText={handleTextChange}
                style={[styles.searchBar, { color: '#000000' }]}
                keyboardType='default'
              />
              {text.length > 0 && (
                <TouchableOpacity
                  onPress={() => { setText(''), Keyboard.dismiss() }}
                  style={{ padding: 10 }}>
                  <FontAwesome
                    name="times"
                    size={18}
                    color={theme.colors[appearance].grey9}
                  />
                </TouchableOpacity>
              )}
            </View>
            {textError ? (
              <Text style={{ color: 'red', marginBottom: 10 }}>
                {textError}
              </Text>
            ) : null}
          </View>

          <TouchableOpacity
            style={styles?.footerContainerSuggestion}
            onPress={onSubmitButton}
            disabled={isLoading}>
            {isLoading ? (
              <ActivityIndicator
                size={'small'}
                color={colorSet.secondaryBackground}
              />
            ) : (
              <Text style={styles?.fotterTitle}>Submit</Text>
            )}
          </TouchableOpacity>

          <TouchableOpacity
            style={styles?.footerContainerSuggestion}
            onPress={onGoBackButton}
            disabled={isLoading}>
            <Text style={styles?.fotterTitle}>Go back</Text>
          </TouchableOpacity>
        </View>
      </TouchableWithoutFeedback>
    </SafeAreaView>`

please help me if any one find the solution : 
@ftroisi @cipolleschi @mobinni @matinzd @alfonsocj @johnnywang

**Comment by kkx64** [comment]
I just encountered this, with `newArchEnabled: false` on Expo 52, we haven't enabled new arch because of a lot of bugs with some of the components. This shouldn't be happening in the first place

**Comment by 1Jesper1** [comment]
Yes, had the same problem using KeyboardAvoidingView inside a ScrollView with behavior=padding. Old arch. Only happened in Hermes. Removing padding seemed to fix it!
